### PR TITLE
Units - Dunefolk - Falconer branch of Skirmisher line

### PR DESCRIPTION
**Pull request by doofus-01**
First draft, not even close to being ready.  Mostly just planting a flag in the ground for now.
EDIT: Relevant forum thread: https://forums.wesnoth.org/viewtopic.php?p=652466#p652466

### CODE REVIEW

**Comment by doofus-01** [comment]
falconer-demo
The lvl2 vs lvl3 needs more distinction, but this is where it is headed.

**Comment by doofus-01** [comment]
There are now some animations.  They probably need a bit more work, but they are close to what I wanted to do...  except for the animation filtering.  As is, the falconer throws the falcon into the air when she moves next to an enemy.  What I actually want to do is have her throw the bird into the air when "distraction" ability is active (which is supposed to be like backstab, but grants the damage boost to the unit on the other side).  But a nested [filter] in a [filter_adjacent] didn't seem to work.  Maybe I made a dumb mistake, maybe it is something in the engine that needs to be fixed, I'll look at it more next week.
Screenshot_20200308

**Comment by Hejnewar** [comment]
Please dont trouble yourself with this ability. Its power level is extremely high on this unit, even without skirmisher and with few movement points less I would still worry about high roll potential of this ability or just the fact that this faction has skirmishers at all. It really is game winning.

**Comment by CelticMinstrel** [comment]
I do want to point out that, if desired, you could also set separate vision costs for the falconer units. I don't know whether doing so is a good idea, just wanted to make sure the possibility wasn't overlooked.

**Comment by doofus-01** [comment]
> I do want to point out that, if desired, you could also set separate vision costs for the falconer units. I don't know whether doing so is a good idea, just wanted to make sure the possibility wasn't overlooked.

Yeah, I think it makes sense for falcon scouting over water, where the bird can see just fine but the human handler can't move very quickly, if at all.  The animation filter is for cave vs non-cave, as it makes sense that the bird isn't flying around in a cramped cave, so probably vision costs can be incorporated in that too.  The unit specs in general should probably be adjusted for cave vs non-cave, but I'm not sure the best way to do that yet.

I'll probably commit this next week, since there was a dev version just released there will be time to fix anything before next release.

**Comment by doofus-01** [comment]
Since it is now possible to modify weapon specials with abilities, I think reducing the enemy chance-to-hit makes more sense than reducing resistances (not that resistances was the top choice, but defense modification isn't simple).  I think I can get the animations to work with Lua, but they currently don't work.

**Comment by doofus-01** [comment]
I think I've got done most of what needs to be done.  There will need to be some adjustments - I sometimes see animation glitches and the specific stats almost certainly need revision.  I also probably violated something in the lua code.

The best way for anyone to notice any of this is to commit to master, I'll do that within the next few days.

**Comment by nemaara** [comment]
I'll write some new descriptions for these units after you merge.

**Comment by Hejnewar** [comment]
Is it the final version of ability for her? If yes then I will take care of balance. 

Falconer and Sky Hunter ranged attacks are really high given that falcon is executing them. 10/17 is not low at all, Drake Blademaster deals 16 damage with single strike for example. Even if falcon has metal claws it still seems high because elder falcon attack is just 4-3.

File: data/lua/diversion.lua [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        local div_candidates = wesnoth.units.find_on_map({
+                ability = "diversion",
+                {"not", { status = "diversion" }}
+                })
+        -- for those that pass the filter now, change status and fire animation
+        for index, ec_unit in ipairs(div_candidates) do
+                local filter_result = u_pos_filter(ec_unit.id)
+                if filter_result then
+                    ec_unit.status.diversion = true
+                    ec_unit:extract()
```
Comment (gfgtdf) [inline data/lua/diversion.lua]: whats the reason for calling extract() foillowed by a to_map() here=

**Comment by gfgtdf** [comment]
wesnoth lua files are usually tab-indented.

**Comment by gfgtdf** [comment]
okay i pushed directly to your branch, since i think this is the easiest way to show what i meant, feel free to change it it you want.

File: data/lua/diversion.lua (hunk 1 above)
Comment (doofus-01) [reply data/lua/diversion.lua]: Not sure, I was just following what was in the petrify/unpetrify tag.

**Comment by doofus-01** [comment]
> okay i pushed directly to your branch, since i think this is the easiest way to show what i meant, feel free to change it it you want.

Seems to work, I've got no problem with it.  Thanks!

**Comment by doofus-01** [comment]
Damage and descriptions can certainly be changed, I didn't invest anything in the current ones.

**Comment by newfrenchy83** [comment]
@doofus-01 when i using diversion on target, if the enemy in opposite side attack instead of Dunefolk Falconer, the cth also reduced to 15%, it is normal or not?

**Comment by doofus-01** [comment]
I think so, it makes sense to me at least.  I didn't put any sort of offense/defense filter in there.

**Comment by newfrenchy83** [comment]
I will reformulate my question, if an A unit with the diversion ability and an Allied C unit surround an enemy B unit, if the combat takes place between B and C instead of A and B cth of B is also reduced by 15%; is this wanted or should cth be reduced only if A and B collide?

**Comment by doofus-01** [comment]
It's functioning as intended.  It makes more sense to me this way.

**Pull request merged**